### ISSUE TITLE
Multus and Calico CNIs have failures with conformance tests

### ISSUE THREAD

**Issue by rancher-max** [rancher/rke2#3010]
**Environmental Info:**
RKE2 Version: 

v1.24.1-rc1+rke2r1, v1.23.7-rc1+rke2r1, v1.22.10-rc1+rke2r1

Node(s) CPU architecture, OS, and Version: 

Saw on sles15sp3, ubuntu 20.04 LTS, and sle micro

Cluster Configuration:

3 servers, 1 agent

**Describe the bug:**

8 tests fail with calico:
```
[sig-network] DNS should provide /etc/hosts entries for the cluster [LinuxOnly] [Conformance]
[sig-cli] Kubectl client Update Demo should scale a replication controller  [Conformance]
[sig-cli] Kubectl client Update Demo should create and stop a replication controller  [Conformance]
[sig-apps] ReplicationController should serve a basic image on each replica with a public image  [Conformance]
[sig-network] DNS should provide DNS for the cluster  [Conformance]
[sig-network] DNS should provide DNS for ExternalName services [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate configmap [Conformance]
[sig-storage] Projected configMap should be consumable from pods in volume with mappings as non-root [NodeConformance] [Conformance]
```
24 tests fail with multus. This taken from a setup that was `multus,canal`:
```
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate custom resource with pruning [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should unconditionally reject operations on fail closed webhook [Conformance]
[sig-storage] ConfigMap should be consumable from pods in volume with defaultMode set [LinuxOnly] [NodeConformance] [Conformance]
[sig-node] PreStop should call prestop when killing a pod  [Conformance]
[sig-api-machinery] Aggregator Should be able to support the 1.17 Sample API Server using the current Aggregator [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should be able to deny attaching pod [Conformance]
[sig-network] DNS should provide DNS for pods for Subdomain [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should not be able to mutate or prevent deletion of webhook configuration objects [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate configmap [Conformance]
[sig-network] DNS should provide /etc/hosts entries for the cluster [LinuxOnly] [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing validating webhooks should work [Conformance]
[sig-network] DNS should provide DNS for ExternalName services [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate pod and apply defaults after mutation [Conformance]
[sig-apps] ReplicationController should serve a basic image on each replica with a public image  [Conformance]
[sig-apps] ReplicaSet should serve a basic image on each replica with a public image  [Conformance]
[sig-network] DNS should provide DNS for the cluster  [Conformance]
[sig-cli] Kubectl client Update Demo should scale a replication controller  [Conformance]
[sig-network] DNS should provide DNS for pods for Hostname [LinuxOnly] [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should be able to deny custom resource creation, update and deletion [Conformance]
[sig-cli] Kubectl client Guestbook application should create and stop a working application  [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] patching/updating a validating webhook should work [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate custom resource [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing mutating webhooks should work [Conformance]
[sig-cli] Kubectl client Update Demo should create and stop a replication controller  [Conformance]
```

**Steps To Reproduce:**

- Installed RKE2 with the specified cni and with ingress disabled:
```
# /etc/rancher/rke2/config.yaml
cni: multus,canal
disable: 
  - rke2-ingress-nginx
```
- run conformance tests. Example for v1.22.10: `sonobuoy run --kubernetes-version=v1.22.10 --mode=certified-conformance`

**Expected behavior:**

All tests should pass

**Actual behavior:**

Multiple failures

**Additional context / logs:**
See two example full results from sonobuoy with these failures (and ingress because in this case I did not disable rke2-ingress-nginx).
Calico: [202206011914_sonobuoy_2a2c2e69-96b1-4590-9958-6158d8af0a49.tar.gz](https://github.com/rancher/rke2/files/8819381/202206011914_sonobuoy_2a2c2e69-96b1-4590-9958-6158d8af0a49.tar.gz)
Multus,canal: [202206011915_sonobuoy_d09a7f27-b624-425d-b35c-1ed2e791bd4d.tar.gz](https://github.com/rancher/rke2/files/8819379/202206011915_sonobuoy_d09a7f27-b624-425d-b35c-1ed2e791bd4d.tar.gz)

**Comment by rancher-max** [comment]
In another two runs, these were the fails:
```
RHEL 8.5 cni: calico

Failed tests: 9
[sig-network] Ingress API should support creating Ingress API operations [Conformance]
[sig-network] DNS should provide DNS for the cluster  [Conformance]
[sig-apps] ReplicationController should serve a basic image on each replica with a public image  [Conformance]
[sig-network] Networking Granular Checks: Pods should function for intra-pod communication: udp [NodeConformance] [Conformance]
[sig-network] DNS should provide DNS for ExternalName services [Conformance]
[sig-cli] Kubectl client Update Demo should create and stop a replication controller  [Conformance]
[sig-apps] ReplicaSet should serve a basic image on each replica with a public image  [Conformance]
[sig-node] PreStop should call prestop when killing a pod  [Conformance]
[sig-cli] Kubectl client Update Demo should scale a replication controller  [Conformance]

RHEL 8.5 cni: multus,cilium

Failed tests: 18
[sig-cli] Kubectl client Update Demo should scale a replication controller  [Conformance]
[sig-network] DNS should provide DNS for the cluster  [Conformance]
[sig-apps] ReplicaSet should serve a basic image on each replica with a public image  [Conformance]
[sig-network] Ingress API should support creating Ingress API operations [Conformance]
[sig-network] HostPort validates that there is no conflict between pods with same hostPort but different hostIP and protocol [LinuxOnly] [Conformance]
[sig-network] DNS should provide /etc/hosts entries for the cluster [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should be able to deny custom resource creation, update and deletion [Conformance]
[sig-network] Services should be able to switch session affinity for service with type clusterIP [LinuxOnly] [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should be able to deny attaching pod [Conformance]
[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] should mutate custom resource with pruning [Conformance]
[sig-network] DNS should resolve DNS of partial qualified names for services [LinuxOnly] [Conformance]
[sig-network] Proxy version v1 A set of valid responses are returned for both pod and service Proxy [Conformance]
[sig-apps] ReplicationController should serve a basic image on each replica with a public image  [Conformance]
[sig-network] Services should have session affinity work for service with type clusterIP [LinuxOnly] [Conformance]
[sig-storage] EmptyDir volumes should support (non-root,0777,default) [LinuxOnly] [NodeConformance] [Conformance]
[sig-network] Services should have session affinity timeout work for service with type clusterIP [LinuxOnly] [Conformance]
[sig-cli] Kubectl client Update Demo should create and stop a replication controller  [Conformance]
[sig-api-machinery] Aggregator Should be able to support the 1.17 Sample API Server using the current Aggregator [Conformance]hould scale a replication controller  [Conformance]
```

**Comment by manuelbuil** [comment]
I started looking at `[sig-network] DNS should provide DNS for ExternalName services [Conformance]` and I was able to reproduce it in my env by executing: `./sonobuoy run --kubernetes-version=v1.24.1 --e2e-focus="DNS should provide DNS for ExternalName services"`. I have only one node:
```
azureuser@mbuil-suse:~> k get nodes -o wide
NAME         STATUS   ROLES                       AGE    VERSION          INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION                CONTAINER-RUNTIME
mbuil-suse   Ready    control-plane,etcd,master   120m   v1.24.1+rke2r1   10.1.1.5      <none>        openSUSE Leap 15.3   5.3.18-150300.59.63-default   containerd://1.6.4-k3s1
```
Sonobuoy creates a namespace `dns-1485` and inside that namespace a service `dns-test-service-3` and a pod `dns-test-1fa49ee6-8a03-4d41-9bd3-a9eb7d781217`. The pod runs:
```
for i in `seq 1 30`; do dig +short dns-test-service-3.dns-54.svc.cluster.local CNAME > /results/jessie_udp@dns-test-service-3.dns-54.svc.cluster.local; sleep 1; done
```
which basically does a DNS query on `dns-test-service-3.dns-54.svc.cluster.local` every second 30 times and dumps the output on a file.

I am able to see that the service can be queried correctly:
1 - I created a clone of the same pod and I can query the service
2 - By checking the file where the output is dumped, I can see the correct output

When you look at the sonobuoy logs, both in my env and what @rancher-max uploaded, you can observe that the repeating error in the logs is:
```
Jun  2 09:24:14.502: INFO: Unable to read jessie_udp@dns-test-service-3.dns-1485.svc.cluster.local from pod dns-1485/dns-test-1fa49ee6-8a03-4d41-9bd3-a9eb7d781217: the server is currently unable to handle the request (**get pods dns-test-1fa49ee6-8a03-4d41-9bd3-a9eb7d781217**
```
It seems as if it can't find the pod, but the pod exists:
```
azureuser@mbuil-suse:~> k get pods dns-test-1fa49ee6-8a03-4d41-9bd3-a9eb7d781217 -n dns-1485
NAME                                            READY   STATUS             RESTARTS       AGE
dns-test-1fa49ee6-8a03-4d41-9bd3-a9eb7d781217   3/3     Running   9 (25s ago)   6m18s
```
I wonder if sonobuoy is somehow forgetting the namespace

**Comment by manuelbuil** [comment]
```
write-kubeconfig-mode: 644
token: "secret"
cni: calico
disable:
  - rke2-ingress-nginx
egress-selector-mode: disabled
```
This fixes it. It must be something with calico and egress-selector-mode

**Comment by rancher-max** [comment]
Validated this has been fixed in all of v1.24.1-rc3+rke2r1, v1.23.7-rc2+rke2r1, and v1.22.10-rc2+rke2r1
